minio-azure-gateway/azuredeploy.json

```diff
@@ -0,0 +1,427 @@
+      }
+    },
+    "nodecount": {
+      "type": "int",
+      "defaultValue": 4,
+      "metadata": {
+        "description": "Initial node count for AKS user node pool"
+      }
+    },
+    "mincount": {
+      "type": "int",
+      "defaultValue": 4,
+      "metadata": {
+        "description": "Min count for AKS user node pool autoscaling"
+      }
+    },
+    "maxcount": {
+      "type": "int",
+      "defaultValue": 10,
+      "metadata": {
+        "description": "Max count for AKS user node pool autoscaling"
+      }
+    },
+    "linuxSSHKey": {
+      "type": "string",
+      "metadata": {
+        "description": "Public key for SSH authentication"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "clusterAdmin",
+      "metadata": {
+        "description": "Linux admin username for AKS cluster nodes"
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located"
+      },
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/minio-azure-gateway/"
+    },
+    "_artifactsLocationSasToken": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The sasToken required to access artifacts"
+      },
+      "defaultValue": ""
+    },
+    "utcValue": {
+      "type": "string",
+      "defaultValue": "[utcNow()]"
+    }
+  },
+  "functions": [],
+  "variables": {
+    "vnetName": "minio-gateway-vnet",
+    "resgpguid": "[substring(replace(guid(resourceGroup().id), '-', ''), 0, 4)]",
+    "installScriptUri": "[uri(parameters('_artifactsLocation'), concat('scripts/minio.sh', parameters('_artifactsLocationSasToken')))]",
+    "helmValuesUri": "[uri(parameters('_artifactsLocation'), concat('scripts/values.yaml', parameters('_artifactsLocationSasToken')))]",
+    "blobStorageAccountName": "[concat('minio', variables('resgpguid'))]",
+    "blobStorageAccountPrivateEndpointName": "[concat('minio', variables('resgpguid'))]",
+    "blobStorageAccountId": "[resourceId('Microsoft.Storage/storageAccounts', variables('blobStorageAccountName'))]",
+    "blobPublicDNSZoneForwarder": "[concat('.blob.', environment().suffixes.storage)]",
+    "blobPrivateDnsZoneName": "[concat('privatelink', variables('blobPublicDNSZoneForwarder'))]",
+    "blobPrivateDnsZoneId": "[resourceId('Microsoft.Network/privateDnsZones', variables('blobPrivateDnsZoneName'))]",
+    "blobStorageAccountPrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', variables('blobStorageAccountPrivateEndpointName'))]",
+    "blobStorageAccountPrivateEndpointGroupName": "blob",
+    "blobPrivateDnsZoneGroup": "[concat(variables('blobStorageAccountPrivateEndpointName'),'/', variables('blobStorageAccountPrivateEndpointGroupName'), 'PrivateDnsZoneGroup')]",
+    "dsRoleAssignmentId": "[guid(resourceGroup().id, deployment().name, 'dsaccess')]",
```

Comment:
this needs to be idempotent and comprised of user, scope and role, eg.

guid(resourceGroup().id, variables('dsRoleDefinitionId'), resourceId('Microsoft.ManagedIdentity/identities', variables('dsIdentityName'))